src/wallet/wallet.cpp

```diff
@@ -4097,7 +4097,7 @@ bool CWallet::Verify(interfaces::Chain& chain, const WalletLocation& location, b
 
 std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain, const WalletLocation& location, uint64_t wallet_creation_flags)
 {
-    const std::string& walletFile = location.GetName();
+    const std::string& walletFile = (location.GetName() == "" ? "[default wallet]" : location.GetName());
```

Comment:
nit, this seems more readable: 

`const std::string& walletFile = location.GetName().empty() ? "[default wallet]" : location.GetName();`

alternatively, why not just have `[default wallet]` be the thing set in the constructor for `WalletLocation` and just remove the need for this special casing?

Reply:
I was uncomfortable with modifying it in WalletLocation because I don't fully understand all the uses that the 'wallet name' is put it, and it seems like sometimes it's used as a path/filename.

Reply:
Done.